sotodlib/tod_ops/jumps.py

```diff
@@ -182,33 +177,39 @@ def jumpfix_subtract_heights(
         x_fixed: x with jumps removed.
                  If inplace is True this is just a reference to x.
     """
-
-    def _fix(jump_ranges, heights, x_fixed):
-        for j, jump_range in enumerate(jump_ranges):
-            for start, end in jump_range.ranges():
-                _heights = heights[j, start:end]
-                height = _heights[np.argmax(np.abs(_heights))]
-                x_fixed[j, int((start + end) / 2) :] -= height
-
+    orig_shape = x.shape
+    x = np.atleast_2d(x)
+    x = np.ascontiguousarray(x)
     x_fixed = x
     if not inplace:
         x_fixed = x.copy()
-    orig_shape = x.shape
-    x_fixed = np.atleast_2d(x_fixed)
+        x_fixed = np.ascontiguousarray(x_fixed)
```

Comment:
To reduce unnecessary reallocations, make use of `np.asarray`.  To do these two lines, for example, you can just:  `x_fixed = np.asarray(x, order='C', copy=True)`